src/gluonnlp/model/utils.py

```diff
@@ -268,6 +274,41 @@ def _load_vocab(dataset_name, vocab, root):
         assert vocab is not None, 'Must specify vocab if not loading from predefined datasets.'
     return vocab
 
+
+def _load_bert_vocab(dataset_name, bert_vocab, root):
+    if dataset_name:
+        if bert_vocab is not None:
+            warnings.warn('Both dataset_name and vocab are specified. \
+            Loading bert_vocab for dataset. '
+                          'Input "bert_vocab" argument will be ignored.')
+        vocab = _load_pretrained_vocab(dataset_name, root)
```

Comment:
We can pass BERTVocab as an argument to `_load_pretrained_vocab` and [_load_vocab_file](https://github.com/dmlc/gluon-nlp/blob/7b8cc05aff944ec4c62cc4eee70da65d141b55fe/src/gluonnlp/data/utils.py#L311-L314) so that cls.from_json() is called, where the default value for `cls` is `Vocab`. This way we do not need to dump everything to json in order to load from json

Reply:
It seems to be a bit confusing because it was based on the logic of loading the original vocab json file into BERTVocab. I will upload a converted BERTVocab json file and modify it in a more concise way.